vcd/resource_vcd_lb_app_rule.go

```diff
@@ -0,0 +1,220 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdLBAppRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLBAppRuleCreate,
+		Read:   resourceVcdLBAppRuleRead,
+		Update: resourceVcdLBAppRuleUpdate,
+		Delete: resourceVcdLBAppRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLBAppRuleImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the Application Rule is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the Application Rule is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the Application Rule is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Application Rule name",
+			},
+			"script": &schema.Schema{
+				Required: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The script for the application rule. Each line as a separate array element",
+			},
+		},
+	}
+}
+
+func resourceVcdLBAppRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	LBRule, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("unable to expand load balancer app rule: %s", err)
+	}
+
+	createdPool, err := edgeGateway.CreateLBAppRule(LBRule)
+	if err != nil {
+		return fmt.Errorf("error creating new load balancer app rule: %s", err)
+	}
+
+	// We store the values once again because response include pool member IDs
+	err = setLBAppRuleData(d, createdPool)
```

Comment:
but ID settted not in this function. Seems comment isn't correct

Reply:
Indeed. This is coming from the "pancake" copy/paste work...
Removed it at all as it doesn't have much meaning in this case.

Reply:
Don't see the change

Reply:
Failed to push. Now it shouldn't be there.